uniswap/model/parts/uniswap_model.py

```diff
@@ -138,13 +146,44 @@ def s_mechanismHub_UNI(_params, substep, sH, s, _input):
         return s_removeLiquidity_UNI(_params, substep, sH, s, _input)
     return('UNI_supply', s['UNI_supply'])
 
-
 # AUX
 
-def getInputPrice(input_amount, input_reserve, output_reserve, _params):
+def get_input_price(input_amount, input_reserve, output_reserve, _params):
     fee_numerator = _params[0]['fee_numerator']
     fee_denominator = _params[0]['fee_denominator']
     input_amount_with_fee = input_amount * fee_numerator
     numerator = input_amount_with_fee * output_reserve
     denominator = (input_reserve * fee_denominator) + input_amount_with_fee
-    return int(numerator // denominator)
\ No newline at end of file
+    return int(numerator // denominator)
+
+def classifier(eth_delta, token_delta, c_rule):
+    if (token_delta / (10 ** (18-c_rule))).is_integer() or (eth_delta / (10 ** (18-c_rule))).is_integer() :
+      return "Arb"
+    else:
+      return "Arb"
+
+def get_trade_decision(input_amount, output_amount, input_reserve, output_reserve, model_price, real_price, _params):
+    if classifier(input_amount, output_amount, _params[0]['c_rule']) == 'Conv':
+        if model_price >= real_price * (1 - _params[0]['conv_tolerance']):
+            return input_amount
+        else:
+            return 0
+    else:
+        spot_price = (input_reserve + input_amount) / (output_reserve + output_amount) 
+        new_input_amount = get_input_amount(spot_price, input_reserve, output_reserve, _params)
+
+        return input_amount
```

Comment:
Shouldn't it return `new_input_amount` in this case? Or maybe the variable should be called just `input_amount`?